### ISSUE TITLE
Can't turn off debug logging

### ISSUE THREAD

**Issue by bmcallis** [Throne3d/node-irc#34]
Debug logging is always on for bots, regardless of the debug flag. 

It appears that this is the problem code
```
    self.out.debug = function() {
        if (!this.debug) return;
        util.log.apply(util, arguments);
    };
```

`this.debug` is that function, so it's never falsey which means that the log always happens. I believe that checking `self.opt.debug` instead will achieve the desired results, but I'm not sure if that's the "right" fix.

**Comment by Throne3d** [comment]
Oh, huh. I thought I did a test for this somewhere to make sure it didn't randomly debug, but presumably I haven't – oops.

I seem to have a name clash, between the function `debug` and the debug boolean I set in the initialization: https://github.com/Throne3d/node-irc/blob/master/lib/irc.js#L102. It should be a relatively quick fix, and I'll try to get to it soon. Thanks!

**Comment by Throne3d** [comment]
Should be fixed in 3eff8bfabaa8b305828cd6fc37d97d0b1750dba9.
